### PR TITLE
Don't emit approval on transfers.

### PR DESCRIPTION
**Pull request by MoMannn**

### CODE REVIEW

File: contracts/tokens/NFToken.sol [hunk 1]
```diff
@@ -350,7 +350,7 @@ contract NFToken is
   {
     address from = idToOwner[_tokenId];
 
-    clearApproval(from, _tokenId);
+    delete idToApprovals[_tokenId];
```
Comment (mg6maciej) [inline contracts/tokens/NFToken.sol]: How about `if (idToApprovals[_tokenId] != 0) { delete idToApprovals[_tokenId]; }`?

Comment (MoMannn) [reply contracts/tokens/NFToken.sol]: I did some gas estimate tests and the results are:
if statement: 315 
delete: 5320

So this change makes sense. Will update.

**Comment by MoMannn** [comment]
It's an internal function and is tested through transfer, burn and approve tests.

**Comment by lknix** [comment]
We need to start testing single units. This is paramount to good design. It's not very helpful if someone makes a breaking change in one function and all unrelated tests start failing. And then there are edge cases which are left out if we test functions through other functions (no necessarily in this case but in general).

**Comment by mg6maciej** [comment]
If I may, I'd like to suggest a different change in this specific case. Change `clearApproval` to private. This function will never be called from derived contract. `_mint` and `_burn` as internal and useful for derived contracts should be tested and seem to be.

I would also change `removeNFToken` and `addNFToken` to private as this is implementation detail that these parts of code are used in multiple places, but I see how this might be problematic, because `NFTokenEnumerable` overrides these. These two functions will not be useful in derived contract and might lead to bugs in derived contracts if used directly. Most likely `_mint`, `_burn` and `transfer`s are to be called in derived contracts to avoid errors.

**Comment by MoMannn** [comment]
clearApproval change to private makes sense.

There was already a discussion about removeNFToken and addNFToken being internal and we decided to keep it that way but we added a notice in the comments:
`@notice Use and override this function with caution. Wrong usage can have serious consequences.`.
